jormungandr/src/blockchain/storage.rs

```diff
@@ -5,12 +5,14 @@ use crate::{
 use chain_storage::store::{for_path_to_nth_ancestor, BlockInfo, BlockStore};
 use tokio::prelude::future::Either;
 use tokio::prelude::*;
+use tokio::sync::lock::Lock;
 
 pub use chain_storage::error::Error as StorageError;
 
 #[derive(Clone)]
 pub struct Storage {
     inner: NodeStorage,
+    write_lock: Lock<NodeStorage>,
 }
```

Comment:
Can you document the structure and how it is meant to work. There are 2 reference to the same object here, having an idea why and how it is supposed to work.